bluesky/magics.py

```diff
@@ -107,55 +109,203 @@ def ct(self, line):
         self._ensure_idle()
         return None
 
-    positioners = []
+    _positioners = []
+    _detectors = []
+
+    @property
+    def positioners(self):
+        warnings.warn("positioners is deprecated." +
+                      "Please use the newer labels feature")
+        return self._positioners
+
+    @positioners.setter
+    def positioners(self, val):
+        warnings.warn("positioners is deprecated." +
+                      "Please use the newer labels feature")
+        self._positioners = val
+
+    @property
+    def detectors(self):
+        warnings.warn("positioners is deprecated." +
+                      "Please use the newer labels feature")
+        return self._detectors
+
+    @detectors.setter
+    def detectors(self, val):
+        warnings.warn("positioners is deprecated." +
+                      "Please use the newer labels feature")
+        self._detectors = val
+
+
     FMT_PREC = 6
 
+
     @line_magic
     def wa(self, line):
         "List positioner info. 'wa' stands for 'where all'."
         if line.strip():
             positioners = eval(line, self.shell.user_ns)
         else:
             positioners = self.positioners
+        if len(positioners) > 0:
+            _print_positioners(positioners, precision=self.FMT_PREC)
+        else:
+            # new behaviour
+            devices_dict = get_labeled_devices(user_ns=self.shell.user_ns)
+            labels = list(devices_dict.keys())
+            for label in labels:
+                print(label)
+                # ignore the first key
+                if are_positioners(devices_dict[label]):
+                    positioners = [positioner[1] for positioner in devices_dict[label]]
+                    _print_positioners(positioners, precision=self.FMT_PREC,
+                                       prefix=" "*8)
+                else:
+                    _print_devices(devices_dict[label], prefix=" "*8)
+
+def _print_devices(devices, prefix=""):
+    cols = ["Python name", "Ophyd Name"]
+    print(prefix + "{:20s} \t {:20s}".format(*cols))
+    print(prefix + "="*40)
+    for name, obj in devices:
+        print(prefix + "{:20s} \t {:20s}".format(name, str(obj.name)))
+
+def are_positioners(devs):
+    # only true if all are positioners
+    # takes ((name, dev),...) tuple
+    logic = True
```

Comment:
This `logic` loop is clever, but `all(is_positioner(obj) for _, obj in devs)` might be simpler.